test/jdk/java/util/Objects/BasicObjectsTest.java

```diff
@@ -45,6 +45,7 @@ public static void main(String... args) {
         errors += testIsNull();
         errors += testNonNull();
         errors += testNonNullOf();
+        errors += testNewIdentity();
```

Comment:
The javadoc of `Objects::newIdentity` claims `[the returned object] can be used for synchronization`. Would it be useful to add a test that tries to synchronize on the result of `Objects.newIdentity` for that as well?